lib/msf/core/post/file.rb

```diff
@@ -537,8 +549,16 @@ def rename_file(old_file, new_file)
   # @param src_file [String] Remote source file name to copy
   # @param dst_file [String] The name for the remote destination file
   def copy_file(src_file, dst_file)
+    return false if directory?(dst_file) or directory?(src_file)
+    verification_token = Rex::Text.rand_text_alpha_upper(8)
     if session.type == "meterpreter"
-      return (session.fs.file.cp(src_file, dst_file).result == 0)
+      begin
+        return (session.fs.file.cp(src_file, dst_file).result == 0)
+      rescue # when the source file is not present meterpreter will raise an error
```

Comment:
Can we refine this to rescue the more specific exception that's raised by Meterpreter in the event that the source file doesn't exist?

Reply:
okay cool, I'll try :+1: